crates/node/src/lib.rs

```diff
@@ -540,6 +550,45 @@ impl Node {
         Ok(outcome)
     }
 
+    async fn substitute_aliases_in_payload(
+        &self,
+        context_id: ContextId,
+        payload: Vec<u8>,
+        aliases: &[Alias<PublicKey>],
+    ) -> Result<Vec<u8>, CallError> {
+        if aliases.is_empty() {
+            return Ok(payload);
+        }
+
+        let mut result = Vec::with_capacity(payload.len());
+        let mut remaining = &payload[..];
+
+        for alias in aliases {
+            let needle_str = format!("{{{alias}}}");
+            let needle = needle_str.into_bytes();
+
+            while let Some(pos) = memmem::find(remaining, &needle) {
+                result.extend_from_slice(&remaining[..pos]);
+
+                let public_key = self
+                    .ctx_manager
+                    .resolve_alias(alias.clone(), Some(context_id))
+                    .map_err(|_| CallError::InternalError)?
+                    .ok_or_else(|| CallError::AliasResolutionFailed {
+                        alias: alias.clone(),
+                    })?;
```

Comment:
you should never need to explicitly clone an alias, deref it to `Copy`

Reply:
Alright!